poetry/packages/vcs_dependency.py

```diff
@@ -71,6 +71,11 @@ def base_pep_508_name(self):  # type: () -> str
 
         requirement += " @ {}+{}@{}".format(self._vcs, self._source, self.reference)
 
+        # add trailing space if 'extra' field is specified between URL
```

Comment:
The `base_pep_508_name` property should return the base name without any extra space since this is the pre-marker part of a dependency.

This code should instead by but inside the `to_pep_508()` method.